testsuite/src/it/java/io/quarkus/ts/startstop/ArtifactGeneratorTest.java

```diff
@@ -0,0 +1,255 @@
+import static io.quarkus.ts.startstop.utils.Logs.parseStartStopTimestamps;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+public class ArtifactGeneratorTest {
+
+    private static final Logger LOGGER = Logger.getLogger(ArtifactGeneratorTest.class.getName());
+
+    public void testRuntime(TestInfo testInfo, String[] extensions, boolean warmUp) throws IOException, InterruptedException {
+        Process pA = null;
+        File buildLogA = null;
+        File runLogA = null;
+        String cn = testInfo.getTestClass().get().getCanonicalName();
+        String mn = testInfo.getTestMethod().get().getName();
+        File appBaseDir = new File(getArtifactGeneBaseDir());
+        File appDir = new File(appBaseDir, Apps.GENERATED_SKELETON.dir);
+        String logsDir = appBaseDir.getAbsolutePath() + File.separator + Apps.GENERATED_SKELETON.dir + "-logs";
+        String extensionsStr = String.join(",", extensions);
+        String[] generatorCmd = ArrayUtils.addAll(MvnCmds.GENERATOR.mvnCmds[0], "-Dextensions=" + extensionsStr + "");
+        URLContent skeletonApp = Apps.GENERATED_SKELETON.urlContent;
+
+        if (warmUp) {
+            LOGGER.info(mn + ": Warming up setup: " + String.join(" ", generatorCmd));
+        } else {
+            LOGGER.info(mn + ": Testing setup: " + String.join(" ", generatorCmd));
+        }
+
+        try {
+            // Cleanup
+            cleanDir(appDir.getAbsolutePath(), logsDir);
+            Files.createDirectories(Paths.get(logsDir));
+
+            // Build
+            buildLogA = new File(logsDir + File.separator + (warmUp ? "warmup-artifact-build.log" : "artifact-build.log"));
+            ExecutorService buildService = Executors.newFixedThreadPool(1);
+            buildService.submit(new Commands.ProcessRunner(appBaseDir, buildLogA, generatorCmd, 20));
+            long buildStarts = System.currentTimeMillis();
+            buildService.shutdown();
+            buildService.awaitTermination(30, TimeUnit.MINUTES);
+            long buildEnds = System.currentTimeMillis();
+
+            assertTrue(buildLogA.exists());
+            checkLog(cn, mn, Apps.GENERATED_SKELETON, MvnCmds.GENERATOR, buildLogA);
+
+            // Config, see app-generated-skeleton/README.md
+            String appPropsSrc = BASE_DIR + File.separator + Apps.GENERATED_SKELETON.dir + File.separator + "application.properties";
+            String appPropsDst = appDir + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "application.properties";
+            Files.copy(Paths.get(appPropsSrc),
+                    Paths.get(appPropsDst), StandardCopyOption.REPLACE_EXISTING);
+
+            // Run
+            LOGGER.info("Running...");
+            runLogA = new File(logsDir + File.separator + (warmUp ? "warmup-dev-run.log" : "dev-run.log"));
+            pA = runCommand(MvnCmds.DEV.mvnCmds[0], appDir, runLogA);
```

Comment:
Why did you go with dev mode and not with building jar ?
I'm looking for reasoning.